Github actions

Scot3004 opened the pull request:
Run automated checks in github actions, all check should pass in github

bandophahita commented on pyproject.toml:
```diff
@@ -0,0 +1,91 @@
+
+# DEVELOPER INSTALL INSTRUCTIONS (assuming current directory is at the top of the project) :
+# Poetry:
+#   poetry install --extras dev
```
My fault for not fixing the original comment in screenpy and screenpy_selenium.  This really SHOULD say `poetry install --extras dev_all`

Scot3004 committed:
Github actions
Add workflows for test, poetry and tests
Add pyproject.toml
Add poetry.lock
ignore_missing_imports in allure

Scot3004 commented:
I don't know why the checks are not being displayed here but, the jobs are running
https://github.com/Scot3004/screenpy_adapter_allure/actions

All the actions are passing now
![image](https://user-images.githubusercontent.com/1175402/230427396-4b59892f-14f5-4132-b27d-9aeeae2c04c9.png)

bandophahita commented:
> I don't know why the checks are not being displayed here...

They are but the UX is a little confusing at first.  The actions for Lint, Tests, and Poetry are under each run.  The run names are auto generated based on what branch they occur in.   
<img width="1269" alt="image" src="https://user-images.githubusercontent.com/4211287/230434114-35bbfb20-4fb2-46cf-b536-0d0041d0623c.png">

 And then if you click on a run, you'll see each environment job.  
<img width="965" alt="image" src="https://user-images.githubusercontent.com/4211287/230434929-83c9c5e6-30fc-42e2-8245-2329772454a6.png">

Click on one of the environment jobs, you'll see the actual break-down of the job.
![image](https://user-images.githubusercontent.com/4211287/230435571-ce72e419-a686-4178-b248-6d032f113de0.png)

Scot3004 commented:
Correct, but when a PR is created it should display the result of the actions here 
![image](https://user-images.githubusercontent.com/1175402/230439581-8619e21c-c2c6-46df-bbcc-7bd647c329a4.png)

Scot3004 commented:
Fun fact, The action Test passed at the first try, the other, required some tune up and learn new stuff like poetry and the pyproject.toml 😂

perrygoy commented on pyproject.toml:
```diff
@@ -0,0 +1,91 @@
+version = "4.0.2"
+description = "An adapter for the microphone of ScreenPy's Narrator which sends logs to Allure."
+authors = ["Perry Goy <perry.goy@gmail.com>"]
+maintainers = ["Gabe Langton", "Marcel Wilson"]
```
Hey @Scot3004, wanna add your name to this list? 😉

Scot3004 replied:
Yes 😄

perrygoy commented on pyproject.toml:
```diff
@@ -0,0 +1,91 @@
+
+[tool.poetry.dependencies]
+python = "^3.8"
+PyHamcrest = ">=2.0.0"
+typing-extensions = ">=4.1.1"
```
I think we can get rid of these two lines; they're dependencies for ScreenPy but not for this adapater.

perrygoy commented on pyproject.toml:
```diff
@@ -0,0 +1,91 @@
+[tool.black]
+target-version = ['py311']
```
Oh, that reminds me—can you add py311 to the `tox.ini` env list in this PR? 🙏

perrygoy reviewed (commented):
I have a few minor things to consider, any changes would dismiss my approval—so i'm just commenting for now!

Could the tests not be showing up for this PR because they aren't merged into `trunk` yet? We'll have to see if the next PR after this one is missing them, too.

bandophahita commented:
> Correct, but when a PR is created it should display the result of the actions here

Ahh... I think that might be due to the fact we haven't added the actions to the trunk yet. Seem to recall something like that happening when I added them to my branch (for screenpy)  before we merged the first time.  

Edit: Confirmed.  The first time the github actions were added to the branch the PR didn't show the checks.  I've never looked into a way around that.

Scot3004 committed:
Update configuration files
Python 3.11 added to tox.ini, removed python 3.7
Updated poetry lock file
Add Sergio Orozco to maintainer
__version__.py using importlib
remove unused pyhamcrest and typing-extensions

Scot3004 commented:
Added a new commit, note I added importlib.metadata to have a single source of truth, because the values are the same than the ones on pyproject.toml

perrygoy reviewed (approved):
Thanks for the quick fixes, @Scot3004—looks good to me!
